addons/mail/models/mail_render_mixin.py

```diff
@@ -30,6 +30,13 @@ def format_datetime(env, dt, tz=False, dt_format='medium', lang_code=False):
     except babel.core.UnknownLocaleError:
         return dt
 
+
+def format_time(env, time, tz=False, time_format='medium', lang_code=False):
+    try:
+        return tools.format_time(env, time, tz=tz, time_format=time_format, lang_code=lang_code)
+    except babel.core.UnknownLocaleError:
+        return time
```

Comment:
Though this is likely the last one, since it's also the first one maybe the try/except bits could be extracted into a common wrapper and the number of bespoke functions reduced from 3 to 1?

Reply:
Like this?

```
def format_date_dt_time(type_to_format, env, date, tz=False, dt_format = False, lang_code=False):
    try:
        if type_to_format == 'date':
            return tools.format_date(env, date, date_format=dt_format, lang_code=lang_code)
        elif type_to_format == 'datetime':
            return tools.format_datetime(env, date, tz=tz, dt_format= dt_format or 'medium', lang_code=lang_code)
        elif type_to_format == 'time':
            return tools.format_time(env, date, tz=tz, time_format = dt_format or 'medium', lang_code=lang_code)
        else:
            return False
    except babel.core.UnknownLocaleError:
        return date
```

Reply:
I think this is a bad idea ... I don't like magic methods, finally you don't understand what it does. Please keep 3 separated methods. At least we understand what they do.